mzLib/Proteomics/Protein.cs

```diff
@@ -15,9 +15,12 @@ public Protein(string sequence, string accession, string name, string full_name,
             FullName = full_name;
             IsDecoy = isDecoy;
             IsContaminant = isContaminant;
+            ProteolysisProducts = new List<ProteolysisProduct>();
+            OneBasedPossibleLocalizedModifications = new Dictionary<int, List<Modification>>();
+            GoTerms = new List<GoTerm>();
```

Comment:
- This solves the problem of null reference exceptions in trying to write XMLs from data read from a FASTA originally. 
- These are not recorded in the FASTAs we read currently, but they might be in the future. 
- Intuitively, there is no reason why these proteins shouldn't contain these properties, so empty sets are more appropriate than null ones.